### PR TITLE
Improve loading of files

### PR DESCRIPTION
**Pull request by MitjaBezensek**
This is a followup to #4502 that makes sure we correctly migrate the data.

### Change type

- [ ] `bugfix`
- [x] `improvement`
- [ ] `feature`
- [ ] `api`
- [ ] `other`

### Release notes

- Improve loading of files.

### CODE REVIEW

File: packages/tldraw/src/lib/utils/tldr/file.ts [hunk 1]
```diff
@@ -148,9 +133,10 @@ export function parseTldrawJsonFile({
 	// records. lets create a store with the records and migrate it to the
 	// latest version
 	let migrationResult: MigrationResult<SerializedStore<TLRecord>>
+	let storeSnapshot: SerializedStore<TLRecord>
 	try {
 		const records = pruneUnusedAssets(data.records as TLRecord[])
-		const storeSnapshot = Object.fromEntries(records.map((r) => [r.id, r]))
+		storeSnapshot = Object.fromEntries(records.map((r) => [r.id, r]))
 		migrationResult = schema.migrateStoreSnapshot({ store: storeSnapshot, schema: data.schema })
```
Comment (ds300) [inline packages/tldraw/src/lib/utils/tldr/file.ts:140]: i think you can omit this line since it will be done again during createTLStore

**Comment by ds300** [comment]
reading this PR made me realize the actual reason why the old v2 files were broken.

The problem is the `upgradeSchema` helper, which didn't create a version entry for the store schema migrations, which had the logic to delete the old record types.

This is the real fix 

image

My bad, I should have figured this out earlier.

**Review by ds300 — APPROVED** [review]
if we also apply the changes in my other comment this will be good to go

File: packages/tldraw/src/lib/utils/tldr/file.ts (hunk 1 above)
Comment (MitjaBezensek) [reply packages/tldraw/src/lib/utils/tldr/file.ts:140]: I does allow us to have a separate error message though. Not sure if we need it though 🤷‍♂️

**Pull request merged**